File: Flutter/flutter_application_1/lib/src/models/reserva.dart

import 'dart:ffi';

import 'package:flutter_application_1/src/models/cliente.dart';
import 'package:flutter_application_1/src/models/habitacion.dart';

class Reservas{
  List<Reserva> items = [];
  Reservas();
  Reservas.fromJsonList(List<dynamic> jsonList) {
    if (jsonList == null) return;
    for (var item in jsonList) {
      final reserva = new Reserva.fromJsonMap(item);
      items.add(reserva);
    }
  }
}

class Reserva{
  int? idReserva;
  String? fechaIni;
  String? fechaFin;
  double? importe;
  String? estado;
  String? dniCliente;
  String? caracteristicas;
  Cliente? cliente;
  Habitacion? habitacion;

  Reserva({
    required this.idReserva,
    required this.fechaIni,
    required this.fechaFin,
    required this.importe,
    required this.estado,
    required this.dniCliente,
    required this.caracteristicas            
  });

  Reserva.fromJsonMap(Map<String, dynamic> json){
    idReserva = json['idReserva'];
    fechaIni = json['fechaIni'] ;
    fechaFin = json['fechaFin'] ;
    importe = json['importe'];
    estado = json['estado'];
    dniCliente = json['dniCliente'];
    caracteristicas = json['caracteristicas'];

    if(json['cliente'] != null){
      cliente = Cliente.fromJsonMap(json['cliente']);
    }

    if(json['habitacion'] != null){
        habitacion = Habitacion.fromJsonMap(json['habitacion']);
    }
  }
}

File: .NET/Proyecto/Prueba1/proyectoFinal/Reserva.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace proyectoFinal
{
    class Reserva
    {
        long idReserva;
        DateTime fechaIni;
        DateTime fechaFin;
        float importe;
        String dniCliente;
        long numHabi;
        String estado;

        public Reserva(long idReserva, DateTime fechaIni, DateTime fechaFin, float importe, string dniCliente, long numHabi, string estado)
        {
            this.idReserva = idReserva;
            this.fechaIni = fechaIni;
            this.fechaFin = fechaFin;
            this.importe = importe;
            this.dniCliente = dniCliente;
            this.numHabi = numHabi;
            this.estado = estado;
        }

        public long IdReserva { get => idReserva; set => idReserva = value; }
        public DateTime FechaIni { get => fechaIni; set => fechaIni = value; }
        public DateTime FechaFin { get => fechaFin; set => fechaFin = value; }
        public float Importe { get => importe; set => importe = value; }
        public string DniCliente { get => dniCliente; set => dniCliente = value; }
        public long NumHabi { get => numHabi; set => numHabi = value; }
        public string Estado { get => estado; set => estado = value; }

        public override string ToString()
        {
            String toString = "" +
                "idReserva:" + idReserva +
                "dniCliente:" + dniCliente +
                "numHabi:" + numHabi +
                "fechaIni:" + fechaIni +
                "fechaFin:" + fechaIni +
                "importe:" + importe +
                "estado" + estado;
            return toString;
        }

    }
}
